### PR TITLE
Document SCRAM authentication upgrade/downgrade

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes DOC-6319

Summary of changes:

- Update the 'Troubleshoot Common Problems' page with instructions for downgrading from SCRAM auth to bcrypt password hashing. While there, also:

  - Break out the SCRAM high CPU content into sections for easier scanning and linking

  - Add some links to the 'Connection Pooling' and 'Transactions' pages where those things are mentioned

  - Add the detail that if the connection pooling software is configured to close idle connections eagerly, this can also contribute to high CPU

- Update the 'SASL/SCRAM-SHA-256 Secure Password-based Authentication' page with specific upgrade instructions showing the SQL statements to type in

### CODE REVIEW

File: v22.2/error-handling-and-troubleshooting.md [hunk 1]
```diff
@@ -71,18 +71,50 @@ For more information about how to troubleshoot cluster-level issues, see [Troubl
...
++ [Overview](#overview)
++ [Mitigation steps while keeping SCRAM enabled](#mitigation-steps-while-keeping-scram-enabled)
++ [Downgrade from SCRAM authentication](#downgrade-from-scram-authentication)
 
-1. SCRAM Authentication is enabled on the cluster.
-1. The client driver's connection pool has no defined maximum number of connections.
-1. The client application issues transactions concurrently.
+#### Overview
+
+When [SASL/SCRAM-SHA-256 Secure Password-based Authentication](security-reference/scram-authentication.html) (SCRAM Authentication) is enabled on a cluster, some additional CPU load is incurred on client applications, which are responsible for handling SCRAM hashing. It's important to plan for this additional CPU load to avoid performance degradation, CPU starvation, and [connection pool](connection-pooling.html) exhaustion on the client. For example, the following set of circumstances can exhaust the client application's resources:
```
Comment (rafiss) [inline v22.2/error-handling-and-troubleshooting.md:80]: one thing we saw in a support ticket is that connection latency increased, even though application CPU usage was still fine. perhaps we can mention that as another possible symptom

File: v22.2/error-handling-and-troubleshooting.md [hunk 2]
```diff
@@ -71,18 +71,50 @@ For more information about how to troubleshoot cluster-level issues, see [Troubl
...
+    ~~~
+
+1. Change the user password encryption algorithm to [bcrypt](https://en.wikipedia.org/wiki/Bcrypt) by setting the [`server.user_login.password_encryption` cluster setting](cluster-settings.html#setting-server-user-login-password-encryption) to `crdb-bcrypt`:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ sql
+    SET server.user_login.password_encryption='crdb-bcrypt'
+    ~~~
+
+1. For each [SQL user](create-user.html) in the system, run [`ALTER USER {user} .. WITH PASSWORD`](alter-user.html#change-a-users-password) to encode the user's password using bcrypt:
```
Comment (rafiss) [inline v22.2/error-handling-and-troubleshooting.md:112]: we could mention that the password can be identical to the current password

**Review by rafiss — APPROVED** [review]
lgtm! i had two optional nits

File: v22.2/error-handling-and-troubleshooting.md (hunk 1 above)
Comment (rmloveland) [reply v22.2/error-handling-and-troubleshooting.md:82]: Added some text about this!

File: v22.2/error-handling-and-troubleshooting.md (hunk 2 above)
Comment (rmloveland) [reply v22.2/error-handling-and-troubleshooting.md]: Added this info!

**Comment by rmloveland** [comment]
> can we add a note stating that: SCRAM does incur additional round trips during authentication which can add latency to the initial connection.

@dikshant I added some text about this

**Review by ianjevans — APPROVED** [review]

**Pull request merged**